src/olympia/addons/forms.py

```diff
@@ -53,18 +53,16 @@ def clean_addon_name(name, instance=None, addon_type=None):
     if not addon_type:
         addon_type = instance.type
 
-    matches = reverse_name_lookup(name, addon_type)
-
-    if matches is not None:
-        addon_id = matches.keys()[0]
+    addon_id = reverse_name_lookup(name, addon_type)
 
+    if addon_id is not None:
         if addon_id and (not instance or addon_id != instance.id):
```

Comment:
We got 3 `if`, some of them testing the same thing, and no else. Can we simplify this to make it easier to read ?